components/feature/remotetabs/src/main/java/mozilla/components/feature/remotetabs/RemoteTabsFeature.kt

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.remotetabs
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.browser.storage.sync.RemoteTabsStorage
+import mozilla.components.browser.storage.sync.Tab
+import mozilla.components.browser.storage.sync.TabEntry
+import mozilla.components.concept.sync.Device
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import mozilla.components.service.fxa.manager.ext.withConstellation
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+
+/**
+ * A feature that listens to the [BrowserStore] changes to update the local remote tabs state
+ * in [RemoteTabsStorage].
+ */
+@ExperimentalCoroutinesApi
+class RemoteTabsFeature(
+    private val accountManager: FxaAccountManager,
+    private val store: BrowserStore,
+    private val tabsStorage: RemoteTabsStorage = RemoteTabsStorage()
+) : LifecycleAwareFeature {
+    private var scope: CoroutineScope? = null
+
+    override fun start() {
+        // TODO: Do we need to `store` the current state right away? Or will the flow be executed on start?
```

Comment:
> Or will the flow be executed on start?

Yes.

You might want to also filter with `ifChanged` so that you can observe changes to only some parts of the store and not _all_ of it.